src/index.spec.ts

```diff
@@ -15,17 +15,23 @@ import Module = require('module')
 import { PassThrough } from 'stream'
 import * as getStream from 'get-stream'
 
-function exec (cmd: string, callback: (err: ExecException | null, stdout: string, stderr: string) => void): ChildProcess
-function exec (cmd: string, opts: ExecOptions, callback: (err: ExecException | null, stdout: string, stderr: string) => void): ChildProcess
-function exec (cmd: string, opts: ExecOptions | ((err: ExecException | null, stdout: string, stderr: string) => void), callback?: (err: ExecException | null, stdout: string, stderr: string) => void) {
-  const _opts = typeof opts === 'function' ? {} : opts
-  const _callback = typeof opts === 'function' ? opts : callback
-  return childProcessExec(cmd, {
-    cwd: TEST_DIR,
-    ..._opts
-  }, _callback)
+type TestExecReturn = { stdout: string, stderr: string, err: null | ExecException }
+function exec (cmd: string, opts: ExecOptions = {}): Promise<TestExecReturn> & { child: ChildProcess } {
+  let childProcess!: ChildProcess
+  return Object.assign(
+    new Promise<TestExecReturn>((resolve, reject) => {
+      childProcess = childProcessExec(cmd, {
+        cwd: TEST_DIR,
+        ...opts
+      }, (error, stdout, stderr) => {
+        resolve({ err: error, stdout, stderr })
```

Comment:
Nit: It's probably more ergonomic to use as `err ? reject(err) : resolve({ stdout, stderr })`.

Reply:
Yeah, that's fair.  There are a few tests that assert that an error *does* happen, and I wasn't sure how it would look to catch the error and then assert against it.
```
let err; try {await exec()} catch(e) {err = e} assert(err)
```

Reply:
Makes sense. I mostly use jest now and it has a `.rejects` you can use, forgot what's included with mocha.

Reply:
I want to switch ts-node to a test runner that supports parallel tests.  I'm thinking of switching to ava, and I'm worried that jest installs too much magic into the runtime environment to enable testing ts-node, since we install into `require.extensions` and install source-map-support.  What do you think?